core/table.ts

```diff
@@ -149,10 +155,10 @@ export class Table {
   public dependencies(value: string | string[]) {
     const newDependencies = typeof value === "string" ? [value] : value;
     newDependencies.forEach(d => {
-      const table = this.session.tables[d];
-
-      if (!!table && table.proto.type === "inline") {
-        table.proto.dependencies.forEach(childDep => this.addDependency(childDep));
+      const allResolved = this.session.findActions(d);
+      const resolved = allResolved.length > 0 ? allResolved[0] : undefined;
+      if (!!resolved && resolved instanceof Table && resolved.proto.type === "inline") {
```

Comment:
this is not due to you, but i just realised this code exists and is a bit crazy, i am pretty sure it doesn't work in at least some circumstances! can you add a `// TODO: This code fails to function correctly if the inline table has not yet been attached to the session. This code probably needs to be moved to compile().`?

Reply:
Yes absolutely, in fact I already mentioned this to you as a potential problem in the previous CR:
https://github.com/dataform-co/dataform/pull/352#discussion_r312616391